lib/Models/CatalogItem.js

```diff
@@ -320,13 +325,80 @@ var CatalogItem = function(terria) {
         }
     });
 
+    /**
+     * Gets the CatalogItems current time.
+     *
+     * This property is an observable version of clock.currentTime, they will always have the same value.
+     *
+     * This property is effectively an interface adapter for clock.definitionChanged which changes the structure from
+     * an Event when the current time changes to a knockout property which can be observed.
+     *
+     * @member {JulianDate} The CatalogItems current time.
+     * @memberOf CatalogItem.prototype
+     */
+    knockout.defineProperty(this, 'currentTime', {
+        // This is defined as a property rather then a member to restrict acess to set. Clients can always
+        // `item.clock.currentTime = value` if they need to set, but since this is a copy set would need to copy the
+        // change to item.clock.currentTime.
+        get : function() {
+            return this._currentTime;
+        },
+    });
+
+    /**
+     * Gets the CatalogItems current time as the discrete time that time the catalogItem has information for.
+     * Returns undefined if it is not possible to query the time (i.e. the item doesn't have a clock or canUseOwnClock is false).
+     *
+     * @member {Date} The CatalogItems discrete current time.
+     * @memberOf CatalogItem.prototype
+     */
+    knockout.defineProperty(this, 'discreteTime', {
+        get : function() {
+            if (!defined(this.currentTime) || !defined(this.availableDates) || !defined(this.intervals) || (this.intervals.length === 0)) {
+                return undefined;
+            }
+
+            return timeAtIndex(this, getCurrentIndex(this));
+        },
+    });
+
+    // A property which defines when we are using the terria clock.
+    // We define this as a knockout property so that we can watch for changes to the propery triggered by observables
+    // that it depends on and update the subscription (watcher) on the terria.clock when the state changes.
+    knockout.defineProperty(this, '_mirroringTerriaClock', {
+        get: function() {
+          return this.isEnabled && this.isShown && !this.useOwnClock && defined(this.clock);
```

Comment:
Consider removing the check for `isShown`. That way the workbench controls will continue to update even while it's hidden from the map.

Reply:
Thanks, discussed offline, have removed isShown, see git commit comment for more details c4e72dd40dc563d74e7918474586417bca0e59ae.